### DISCUSSION TITLE
CRC issues reported by wmbusmeters with HASS, nanoCUL and a F90 heat meter

### DISCUSSION THREAD

**Discussion by im0nKeY84** [wmbusmeters/wmbusmeters#1429] [Q&A] [answered]
Hi,

I just wanted to add two heat meters to my HA instance and brought a "nanoCUL USB Stick FTDI CC1101 868MHz M-Bus + Adapter/Wireless M-Bus" on Amazon, which should already have the correct firmware to be used with wmbusmeters (nanoCUL_r568_mbus_c1t1_bufsize300).

I'm using Proxmox for my HA instance and forwarded the the USB device to it. The the serial device has been found in HA and I installed the wmbusmeters HA add-on. The add-on is also successfully finding the serial device, but I get CRC errors reported, on received data:

```
(serial) received binary "62594144343443"
(cul) checkCULFrame "bYAD44C"
(cul) no eol found yet, partial frame
(serial) received binary "35313439383132303733333030303437414233303039303235434238333539424133414143373246433042433038303742333939324437343730323337"
(cul) checkCULFrame "bYAD44C5149812073300047AB3009025CB8359BA3AAC72FC0BC0807B3992D7470237"
(cul) no eol found yet, partial frame
(serial) received binary "33344346353636323731303944314232373736423135323246414337373330323133433832434636353536383537343032303946304144343133364144"
(cul) checkCULFrame "bYAD44C5149812073300047AB3009025CB8359BA3AAC72FC0BC0807B3992D747023734CF56627109D1B2776B1522FAC7730213C82CF655685740209F0AD4136AD"
(cul) no eol found yet, partial frame
(serial) received binary "4443354544364232434641393242323745433839363538364336374637303145433239314535444332373533434537453932414536353138464332463945444433344230343737444534303538423033343932413633453638463539364431384235424146453138353730313536414245323041344135"
(cul) checkCULFrame "bYAD44C5149812073300047AB3009025CB8359BA3AAC72FC0BC0807B3992D747023734CF56627109D1B2776B1522FAC7730213C82CF655685740209F0AD4136ADDC5ED6B2CFA92B27EC896586C67F701EC291E5DC2753CE7E92AE6518FC2F9EDD34B0477DE4058B03492A63E68F596D18B5BAFE18570156ABE20A4A5"
(cul) no eol found yet, partial frame
(serial) received binary "33333236413244444432424241393846394633353941343730443443384344453239364435304241354532463445324332350D0A"
(cul) checkCULFrame "bYAD44C5149812073300047AB3009025CB8359BA3AAC72FC0BC0807B3992D747023734CF56627109D1B2776B1522FAC7730213C82CF655685740209F0AD4136ADDC5ED6B2CFA92B27EC896586C67F701EC291E5DC2753CE7E92AE6518FC2F9EDD34B0477DE4058B03492A63E68F596D18B5BAFE18570156ABE20A4A53326A2DDD2BBA98F9F359A470D4C8CDE296D50BA5E2F4E2C25<0D><0A>"
(cul) checkCULFrame RSSI_RAW=37
(cul) checkCULFrame LQI=22
(wmbus) trimming frame B "AD44C5149812073300047AB3009025CB8359BA3AAC72FC0BC0807B3992D747023734CF56627109D1B2776B1522FAC7730213C82CF655685740209F0AD4136ADDC5ED6B2CFA92B27EC896586C67F701EC291E5DC2753CE7E92AE6518FC2F9EDD34B0477DE4058B03492A63E68F596D18B5BAFE18570156ABE20A4A53326A2DDD2BBA98F9F359A470D4C8CDE296D50BA5E2F4E"
(wmbus) ff b dll crc first 0-126 ddd2 ok
(wmbus) ff b dll crc (calculated 2054) did not match (expected 2f4e) for bytes 128-144!
(cul) dll C1 (frame b) crcs failed check! Ignoring telegram!
(cul) error in received message.
```

I'm also using a F90U3 like mentioned here:

https://github.com/wmbusmeters/wmbusmeters/issues/1376

I'm currently don't know how to investigate further. I already reduced the distance by adding an USB extension cord, but that did not change anything.

I attached wmbusmeters log file from HA.

[c8a990ad_wmbusmeters-ha-addon_2024-12-01T17-26-17.162Z.log](https://github.com/user-attachments/files/17969706/c8a990ad_wmbusmeters-ha-addon_2024-12-01T17-26-17.162Z.log)

Greetings
Sebastian

**Comment by weetmuts** [comment, marked as answer]
I think I should remove nanoCUL as a supported dongle. :-( Its limitation to max 148 byte long wmbus telegrams is just horrible.

Or perhaps print an enormous warning at startup! :-)

Your telegram is most likely too long.

Also the only way to detect that the telegram has been truncated is that the crc error is wrong.....or sometimes an odd number of hex are sent to wmbusmeters, but that generates a proper warning.

So, yeah, buy another dongle. 
 
https://wmbusmeters.github.io/wmbusmeters-wiki/nanoCUL.html

**Comment by im0nKeY84** [comment]
Hi @weetmuts,

thanks for the info. Makes sense, as I used a nanoCUL years ago with FHEM. Any dongle you would recommend?

I found so far the [IMST iU891A-XL](https://shop.imst.de/wireless-solutions/usb-radio-products/89/iu891a-xl-wireless-m-bus-usb-adapter-868-mhz?c=14) and the [AMB8665-M (Würth/Amber Metis-II)](https://www.digikey.de/de/products/detail/w%C3%BCrth-elektronik/2607046213001/8258134), which is quite expensive and hard to get...

Maybe any do-it-yourself projects with an ESP32 or similar?

Greetings
Sebastian

**Reply by weetmuts** [reply]
You can check my personal tests here: https://wmbusmeters.github.io/wmbusmeters-wiki/ANTENNA.html 

Cheapest is of course a generic rtlsdr dongle (not the fancy metallic ones), works fine if you have the cpu cycles to spare and its not too far from the meter.

**Reply by horstvogel** [reply]
> Hi @weetmuts,
> 
> thanks for the info. Makes sense, as I used a nanoCUL years ago with FHEM. Any dongle you would recommend?
> 
> I found so far the [IMST iU891A-XL](https://shop.imst.de/wireless-solutions/usb-radio-products/89/iu891a-xl-wireless-m-bus-usb-adapter-868-mhz?c=14) and the [AMB8665-M (Würth/Amber Metis-II)](https://www.digikey.de/de/products/detail/w%C3%BCrth-elektronik/2607046213001/8258134), which is quite expensive and hard to get...
> 
> Maybe any do-it-yourself projects with an ESP32 or similar?
> 
> Greetings Sebastian

The Link Würth/Amber shows a   WIRELESS M-BUS USB **169MHZ**

**Comment by im0nKeY84** [comment]
Thank you very much for your support. I brought a IMST iU891A-XL and now it is working fine.
